src/braket/circuits/diagram_builders/braket_circuit_drawer.py

```diff
@@ -0,0 +1,402 @@
+        elif gate in ["TOFFOLI", "CCX", "CCNOT"] and len(qubits) == 3:
+            q_ctrl1, q_ctrl2, q_tgt = qubits
+            matrix[q_ctrl1][col] = f"\\ctrl{{{q_tgt - q_ctrl1}}}"
+            matrix[q_ctrl2][col] = f"\\ctrl{{{q_tgt - q_ctrl2}}}"
+            matrix[q_tgt][col] = "\\targ{}"
+
+        # 5) Measurement(s) (MEASURE) on one or more qubits
+        elif gate.startswith("MEASURE"):
+            for q in qubits:
+                matrix[q][col] = "\\meter{}"
+
+        # 6) Generic multi-qubit gates
+        else:
+            top = min(qubits)
+            wires = len(qubits)
+            matrix[top][col] = f"\\gate[wires={wires}]{{{gate}}}"
+            # Leave the rest of the rows “below” empty so quantikz collapses wires
+            for q in qubits[1:]:
+                matrix[q][col] = ""
+
+    # Build the LaTeX lines
+    lines = []
+    lines.append("\\begin{quantikz}")
+    for q in range(n_qubits):
+        if initial_states and q < len(initial_states):
+            label = initial_states[q]
+        else:
+            label = "\\lstick{\\ket{0}}"
+        row_tokens = [label] + matrix[q]
+        line = " & ".join(row_tokens) + " \\\\"
+        lines.append(line)
+    lines.append("\\end{quantikz}")
+
+    latex_code = "\n".join(lines)
+    return latex_code
+
+
+def circuit_drawer(circuit, style="mpl", **kwargs):
+    """
+    High-level function to visualize a Braket Circuit.
+
+    Parameters:
+      - circuit: braket.circuits.Circuit object
+      - style: 'mpl' for Matplotlib, 'latex' for quantikz output
+      - kwargs: any extra keyword arguments are forwarded to draw_circuit_matplotlib
+                (if style='mpl') or to generate_circuit_latex (if style='latex').
+
+    Returns:
+      - If style=='mpl': a tuple (fig, ax) of Matplotlib Figure and Axes.
+      - If style=='latex': a string containing the LaTeX code.
+    """
+    if style == "mpl":
+        return draw_circuit_matplotlib(circuit, **kwargs)
+    elif style == "latex":
+        return generate_circuit_latex(circuit, **kwargs)
+    else:
+        raise ValueError("Unknown style: must be 'mpl' or 'latex'.")
+
+
+if __name__ == "__main__":
+    try:
+        from braket.circuits import Circuit
+        import matplotlib.pyplot as plt
+
+        # Example circuit: H, CNOT, RX(π/2), CCNOT, measure qubits 0,1,2
+        circuit = Circuit().h(0).cnot(0, 1).rx(1, 1.5708).ccnot(0, 1, 2).measure(4)
+
+        # 1) Visualize with Matplotlib
+        fig, ax = draw_circuit_matplotlib(circuit, figsize=(6, 4))
+        plt.show()
+
+        # 2) Generate LaTeX (quantikz)
+        latex = generate_circuit_latex(circuit, initial_states=["|0>", "|0>", "|0>"])
+        print("\n------ LaTeX (quantikz) code: ------\n")
+        print(latex)
+
+    except ImportError:
+        print(
+            "AWS Braket SDK is not installed. To test this example, run:\n"
+            "    pip install amazon-braket-sdk\n"
+        )
+
```

Comment:
I'd remove this section - we don't need the `.py` files to be runnable. You can certainly use it locally as a test script, or move it to a unit test which verifies the matplotlib and LaTeX outputs for this circuit.